rocketpy/Flight.py

```diff
@@ -1563,6 +1575,258 @@ def uDot(self, t, u, postProcessing=False):
 
         return uDot
 
+    def uDotGeneralized(self, t, u, postProcessing=False):
+        """Calculates derivative of u state vector with respect to time when the
+        rocket is flying in 6 DOF motion in space and significant mass variation
+        effects exist. Typical flight phases include powered ascent after launch
+        rail.
+
+        Parameters
+        ----------
+        t : float
+            Time in seconds
+        u : list
+            State vector defined by u = [x, y, z, vx, vy, vz, q0, q1,
+            q2, q3, omega1, omega2, omega3].
+        postProcessing : bool, optional
+            If True, adds flight data information directly to self variables
+            such as self.attackAngle, by default False.
+
+        Returns
+        -------
+        uDot : list
+            State vector defined by uDot = [vx, vy, vz, ax, ay, az,
+            e0Dot, e1Dot, e2Dot, e3Dot, alpha1, alpha2, alpha3].
+        """
+        # Retrieve integration data
+        x, y, z, vx, vy, vz, e0, e1, e2, e3, omega1, omega2, omega3 = u
+
+        # Create necessary vectors
+        # r = Vector([x, y, z])               # CDM position vector
+        v = Vector([vx, vy, vz])  # CDM velocity vector
+        e = [e0, e1, e2, e3]  # Euler parameters/quaternions
+        w = Vector([omega1, omega2, omega3])  # Angular velocity vector
+
+        # Retrieve necessary quantities
+        rho = self.env.density.getValueOpt(z)
+        total_mass = self.rocket.totalMass.getValueOpt(t)
+        total_mass_dot = self.rocket.totalMass.differentiate(t)
+        total_mass_ddot = self.rocket.totalMass.differentiate(t, order=2)
```

Comment:
Did you consider to calculate the total_mass_dot inside the Rocket class, storing it as a Function attribute, and then simply using `getValueOpt()` to retrieve the value at each time it is needed? 

I wonder if the differentiate function could be slowing down the code since the derivative is being calculated inside the simulation loop.

Plus, I also think that the user could want to see the total_mass_dot and total_mass_ddot before running the flight simulation.

Reply:
I think it is a great idea as long as it does not make the simulation slower, which it probably won't. I think this will definitely be changed when we go through with optimizing the speed of the new EOMs

Reply:
Let's keep an eye on it for future PRs!